pupil_src/shared_modules/pupil_detector_plugins/detector_base_plugin.py

```diff
@@ -66,130 +104,135 @@ def init_ui(self):
     def deinit_ui(self):
         self.remove_menu()
 
-    @property
-    def enabled(self):
-        return self._enabled
-
-    @enabled.setter
-    def enabled(self, value):
-        self._enabled = value
-        for elem in self.menu:
-            elem.read_only = not self.enabled
-
     def recent_events(self, event):
-        if not self.enabled:
-            self._recent_detection_result = None
-            return
+        frame = event.get("frame", None)
 
-        frame = event.get("frame")
-        if not frame:
+        if not frame or not self.enabled:
             self._recent_detection_result = None
             return
 
-        frame_size = (frame.width, frame.height)
-        if frame_size != self._last_frame_size:
-            if self._last_frame_size is not None:
-                self.on_resolution_change(self._last_frame_size, frame_size)
-            self._last_frame_size = frame_size
+        # Detect if resolution changed
+        self._update_frame_size_if_changed(new_frame_size=(frame.width, frame.height))
+
+        # Get results of previous detectors
+        # TODO: This is currently used by Pye3D to get the results of the 2D detection
+        previous_detection_results = event.get(EVENT_KEY, [])
 
         detection_result = self.detect(
             frame=frame,
             # TODO: workaround to get 2D data into pye3D for now
-            previous_detection_results=event.get(EVENT_KEY, []),
+            previous_detection_results=previous_detection_results,
         )
 
-        if EVENT_KEY in event:
-            event[EVENT_KEY].append(detection_result)
-        else:
-            event[EVENT_KEY] = [detection_result]
+        # Append the new detection result to the previous results
+        event[EVENT_KEY] = previous_detection_results + [detection_result]
 
+        # Save the most recent detection result for visualization
         self._recent_detection_result = detection_result
 
-    @abc.abstractmethod
-    def detect(self, frame, **kwargs):
+    def on_notify(self, notification):
+        subject_components = notification["subject"].split(".")
+
+        if subject_components[0] == "pupil_detector":
```

Comment:
I feel like we should add a thin layer of sanity checking and wrap the code below into a try/except